src/main/java/org/kainos/ea/db/EmployeeDao.java

```diff
@@ -37,14 +38,75 @@ public List<Employee> getAllEmployees() throws SQLException {
         return employeeList;
     }
 
+    public int createNewEmployee(EmployeeRequest employeeRequest) throws SQLException, DeliveryEmployeeCouldNotBeCreatedException {
+
+        try {
+            Connection c = databaseConnector.getConnection();
+
+            String insertStatement = "INSERT INTO employee (f_name, l_name, salary, bank_acc_num, ni_num) " +
+                    "VALUES (?, ?, ?, ?, ?)";
+
+            PreparedStatement st = c.prepareStatement(insertStatement, Statement.RETURN_GENERATED_KEYS);
+
+            st.setString(1, employeeRequest.getfName());
+            st.setString(2, employeeRequest.getlName());
+            st.setDouble(3, employeeRequest.getSalary());
+            st.setString(4, employeeRequest.getBankAccNum());
+            st.setString(5, employeeRequest.getNiNum());
+
+            st.executeUpdate();
+
+            ResultSet rs = st.getGeneratedKeys();
+
+
+            if (rs.next()) {
+
+                System.out.println("Adding New Employee Delivery.");
+
+                // Add New Create ID To Employee_delivery Table
+                int newlyCreatedIDEmployeeTable = rs.getInt(1);
+                ResultSet rsEmployeeDelivery;
+
+                insertStatement = "INSERT INTO delivery_employee (employee_id) " +
+                        "VALUES (?)";
+
+                st = c.prepareStatement(insertStatement, Statement.RETURN_GENERATED_KEYS);
+
+                st.setInt(1, newlyCreatedIDEmployeeTable);
+
+                rsEmployeeDelivery = st.getGeneratedKeys();
+
+                if (rsEmployeeDelivery != null) {
+                    return newlyCreatedIDEmployeeTable;
+                }
+
+                // Failed to add employee to delivery_employee
+                System.err.println("Delivery Employee Could Not Be Added. Removing From Employee Table.");
+
+                // Need to remove employee from table.
+                String deleteStatement = "DELETE FROM employee WHERE employee_id = ?";
+                st = c.prepareStatement(deleteStatement);
+                st.setInt(1, newlyCreatedIDEmployeeTable);
+                st.executeUpdate();
+                throw new DeliveryEmployeeCouldNotBeCreatedException();
+            }
+
+            // Error - Could not add employee to table.
+            throw new DeliveryEmployeeCouldNotBeCreatedException();
+        } catch (SQLException e) {
+            System.err.println(e.getMessage());
+            throw new SQLException();
```

Comment:
you're throwing the same exception that your catching, did you mean to throw DeliveryEmployeeCouldNotBeCreatedException?

Reply:
nope!